partnership/partnership.go

```diff
@@ -238,52 +244,72 @@ func facilePartnership(jwtData string, policy *models.Policy, product *models.Pr
+		return fmt.Errorf("jwt expired")
+	}
 
-	if claims, ok := token.Claims.(*FacileClaims); ok && token.Valid {
-		log.Println("[facilePartnership] setting person info")
-		person.Name = claims.CustomerName
-		person.Surname = claims.CustomerFamilyName
-		person.Mail = claims.Email
-		birthDate, _ := time.Parse(models.TimeDateOnly, claims.CustomerBirthDate)
-		person.BirthDate = birthDate.Format(time.RFC3339)
-		person.Phone = fmt.Sprintf("+39%s", claims.Mobile)
-		person.Gender = claims.Gender
-		policy.Contractor = person
-		asset.Person = &person
-		policy.OfferlName = "default"
+	log.Println("[facilePartnership] setting person info")
+	person.Name = claims.CustomerName
+	person.Surname = claims.CustomerFamilyName
+	person.Mail = claims.Email
+	birthDate, _ := time.Parse(models.TimeDateOnly, claims.CustomerBirthDate)
+	person.BirthDate = birthDate.Format(time.RFC3339)
+	person.Phone = fmt.Sprintf("+39%s", claims.Mobile)
+	person.Gender = claims.Gender
+	policy.Contractor = person
+	asset.Person = &person
+	policy.OfferlName = "default"
+
+	log.Println("[facilePartnership] setting death guarantee info")
+
+	deathGuarantee := product.Companies[0].GuaranteesMap["death"]
+	deathGuarantee.Value = &models.GuaranteValue{
+		Duration: &models.Duration{
+			Year: claims.Duration,
+		},
+		SumInsuredLimitOfIndemnity: float64(claims.InsuredCapital),
+	}
+	asset.Guarantees = make([]models.Guarante, 0)
+	asset.Guarantees = append(asset.Guarantees, *deathGuarantee)
 
-		log.Println("[facilePartnership] setting death guarantee info")
+	policy.Assets = append(policy.Assets, asset)
+	policy.PartnershipData = claims.ToMap()
+	return err
+}
 
-		deathGuarantee := product.Companies[0].GuaranteesMap["death"]
-		deathGuarantee.Value = &models.GuaranteValue{
-			Duration: &models.Duration{
-				Year: claims.Duration,
-			},
-			SumInsuredLimitOfIndemnity: float64(claims.InsuredCapital),
-		}
-		asset.Guarantees = make([]models.Guarante, 0)
-		asset.Guarantees = append(asset.Guarantees, *deathGuarantee)
+func decryptJwt[T interface{}](jwtData, key string) (T, error) {
+	var claims T
+	object, err := jose.ParseEncrypted(jwtData)
+	if err != nil {
+		log.Printf("[facilePartnership] could not parse jwt - %s", err.Error())
+		return claims, fmt.Errorf("could not decrypt jwt")
+	}
 
-		policy.Assets = append(policy.Assets, asset)
-		policy.PartnershipData = claims.ToMap()
-		return err
+	decryptionKey, err := b64.StdEncoding.DecodeString(key)
+	if err != nil {
+		log.Printf("[facilePartnership] could not decode signing key - %s", err.Error())
+		return claims, fmt.Errorf("could not decrypt jwt")
+	}
+	decrypted, err := object.Decrypt(decryptionKey)
+	if err != nil {
+		log.Printf("[facilePartnership] could not decrypt jwt - %s", err.Error())
+		return claims, fmt.Errorf("could not decrypt jwt")
 	}
 
-	log.Printf("[facilePartnership] could not validate facile partnership JWT - %s", err.Error())
-	return err
+	err = json.Unmarshal(decrypted, &claims)
+	return claims, err
```

Comment:
It seems like a lib function. Either way, I'd update the text in the print statements with the correct function name.

Reply:
Refactored in https://github.com/wopta/goworkspace/pull/141/commits/4f4dbdf2c9332928b46098f5099964a5824d1be6